froide/helper/email_parsing.py

```diff
@@ -269,11 +269,21 @@ def try_decoding(encoded, encoding=None):
     return decoded
 
 
+def check_mail(email: str):
+    # Credit for regex to Anđela Niketić at https://stackabuse.com/python-validate-email-address-with-regular-expressions-regex/
+    regex_mail = r"([-!#-'*+/-9=?A-Z^-~]+(\.[-!#-'*+/-9=?A-Z^-~]+)*|\"([]!#-[^-~ \t]|(\\[\t -~]))+\")@([-!#-'*+/-9=?A-Z^-~]+(\.[-!#-'*+/-9=?A-Z^-~]+)*|\[[\t -Z^-~]*])"
```

Comment:
This regex is too complex to maintain for my taste (and still doesn't capture the complexity of proper email address parsing if you wanted to go there).
Please use the [Django email validator](https://docs.djangoproject.com/en/4.1/ref/validators/#validate-email) instead.

Reply:
Thank you also for this feedback! I used the Django Email validator as you suggested. I recognized that it also catched two cases were the emails in the test data were invalid (according to the django validator):

- In foirequest/tests/testdata/test_mail_05.txt on line 33: "CC: <KÃ¤mmerer.a@zdf.de>, <Biogiani.D@zdf.de>". The test `test_recipient_parsing` fails if the django email validator is used, because the first email-address is filtered. This seems so be a general problem with addresses which have umlaut characters in the local part. From what I found the django EmailValidator does [not support non-ASCII characters](https://code.djangoproject.com/ticket/27029) in the [local part](https://code.djangoproject.com/ticket/31053#comment:4). The [email-validator](https://pypi.org/project/email-validator/) for example has an `allow_smtputf8` option which seems to be not available in the django validator.
- In foirequest/tests/testdata/test_mail_08.txt on line 2: "X-Original-To: \<fragdenstaat class="de"\>\</fragdenstaat\>".

I could additionally change the mail addresses in those two cases in the test data if this makes sense? And would it make sense to use another validator because of the umlaut problem?